File: ruby/raindrops/raindrops.rb
module Raindrops

  RAIN_DROPS = {
    3 => "Pling",
    5 => "Plang",
    7 => "Plong"
  }

  def self.convert(num)
    melody = RAIN_DROPS
      .select { |(factor, sound)| (num % factor).zero? }
      .map { |(factor, sound)| sound }

    melody.empty? ? num.to_s :  melody.join
  end

end

module BookKeeping
  VERSION = 2
end

File: lua/raindrops/raindrops.lua
local sounds = {
  {3, "Pling"},
  {5, "Plang"},
  {7, "Plong"},
}

return function(num)
  local melody = ""

  for _, tuple in ipairs(sounds) do
    local factor, sound = unpack(tuple)
    if num % factor == 0 then
      melody = melody .. sound
    end
  end

  return melody == "" and tostring(num) or melody
end